### ISSUE TITLE
task_run routine is undeclared

### ISSUE THREAD

**Issue by Altai-man** [melezhik/sparrowdo-chef-client#2]
Hi! It seems there was some regression, and it was there in 2019.11 release too.
```
➜  ~ zef install Sparrowdo::Chef::Client
===> Searching for: Sparrowdo::Chef::Client
===> Searching for missing dependencies: Sparrowdo, JSON::Tiny
===> Searching for missing dependencies: Config::Simple, Sparrow6
===> Searching for missing dependencies: File::Directory::Tree, Hash::Merge, YAMLish, JSON::Pretty, Config::INI, Data::Dump
===> Searching for missing dependencies: Test::META, MIME::Base64
===> Searching for missing dependencies: META6, URI, License::SPDX
===> Searching for missing dependencies: JSON::Class:ver<0.0.14+>, JSON::Class:ver<0.0.5+>, JSON::Fast
===> Searching for missing dependencies: JSON::Marshal:ver<0.0.18+>, JSON::Unmarshal:ver<0.08+>
===> Searching for missing dependencies: JSON::Name
===> Testing: JSON::Tiny:ver<1.0>
===> Testing [OK] for JSON::Tiny:ver<1.0>
===> Testing: JSON::Pretty:ver<0.1.0>:auth<github:FROGGS>
===> Testing [OK] for JSON::Pretty:ver<0.1.0>:auth<github:FROGGS>
===> Testing: Config::INI
===> Testing [OK] for Config::INI
===> Testing: Data::Dump:ver<v.0.0.11>:auth<github:tony-o>
===> Testing [OK] for Data::Dump:ver<v.0.0.11>:auth<github:tony-o>
===> Testing: Config::Simple:auth<Sylvain Colinet>
===> Testing [OK] for Config::Simple:auth<Sylvain Colinet>
===> Testing: File::Directory::Tree:auth<labster>
===> Testing [OK] for File::Directory::Tree:auth<labster>
===> Testing: JSON::Fast:ver<0.10>
===> Testing [OK] for JSON::Fast:ver<0.10>
===> Testing: JSON::Name:ver<0.0.4>:auth<github:jonathanstowe>:api<1.0>
===> Testing [OK] for JSON::Name:ver<0.0.4>:auth<github:jonathanstowe>:api<1.0>
===> Testing: JSON::Marshal:ver<0.0.18>:auth<github:jonathanstowe>:api<1.0>
===> Testing [OK] for JSON::Marshal:ver<0.0.18>:auth<github:jonathanstowe>:api<1.0>
===> Testing: JSON::Unmarshal:ver<0.08>
===> Testing [OK] for JSON::Unmarshal:ver<0.08>
===> Testing: JSON::Class:ver<0.0.14>:auth<github:jonathanstowe>:api<1.0>
===> Testing [OK] for JSON::Class:ver<0.0.14>:auth<github:jonathanstowe>:api<1.0>
===> Testing: META6:ver<0.0.23>:auth<github:jonathanstowe>:api<1.0>
===> Testing [OK] for META6:ver<0.0.23>:auth<github:jonathanstowe>:api<1.0>
===> Testing: URI:ver<0.3.0>
===> Testing [OK] for URI:ver<0.3.0>
===> Testing: License::SPDX:ver<3.7.1>:auth<github:jonathanstowe>:api<1.0>
===> Testing [OK] for License::SPDX:ver<3.7.1>:auth<github:jonathanstowe>:api<1.0>
===> Testing: Test::META:ver<0.0.16>:auth<github:jonathanstowe>:api<1.0>
[Test::META] # the following may make some diagnostics from the module itself
===> Testing [OK] for Test::META:ver<0.0.16>:auth<github:jonathanstowe>:api<1.0>
===> Testing: Hash::Merge:ver<1.0.0>:auth<github:scriptkitties>:api<1>
[Hash::Merge] Cannot merge an undefined Hash!
[Hash::Merge]   in method merge at t/02-empty-source.t line 22
[Hash::Merge] Cannot merge an undefined Hash!
[Hash::Merge]   in method merge at t/02-empty-source.t line 22
===> Testing [OK] for Hash::Merge:ver<1.0.0>:auth<github:scriptkitties>:api<1>
===> Testing: MIME::Base64:ver<1.2.1>:auth<github:retupmoca>
===> Testing [OK] for MIME::Base64:ver<1.2.1>:auth<github:retupmoca>
===> Testing: YAMLish:ver<0.0.5>
===> Testing [OK] for YAMLish:ver<0.0.5>
===> Testing: Sparrow6:ver<0.0.11>
===> Testing [OK] for Sparrow6:ver<0.0.11>
===> Testing: Sparrowdo:ver<0.1.1>
===> Testing [OK] for Sparrowdo:ver<0.1.1>
===> Testing: Sparrowdo::Chef::Client:ver<0.0.3>
[Sparrowdo::Chef::Client] ===SORRY!=== Error while compiling /home/koto/.zef/store/sparrowdo-chef-client.git/d446387f109ea8073868c4703fecf70239188630/lib/Sparrowdo/Chef/Client.pm6 (Sparrowdo::Chef::Client)
[Sparrowdo::Chef::Client] Undeclared routine:
[Sparrowdo::Chef::Client]     task_run used at lines 22, 45
===> Testing [FAIL]: Sparrowdo::Chef::Client:ver<0.0.3>
Aborting due to test failure: Sparrowdo::Chef::Client:ver<0.0.3> (use --force-test to override)
```
Currently `task_run` being missed is an issue for installation of various modules, maybe there is a single common reason?

**Comment by melezhik** [comment]
the same reason as for https://github.com/melezhik/sparrowdo-azure-web-cert-module/issues/1 , the last version of Sparrow6, Sparrowdo is not compatible with the module. Again if it urgent, please let me know. I need to apply the same fix for all `Sparrowdo::`  modules.

**Comment by Altai-man** [comment]
Thanks for quickness!
It is not urgent, I am just notifying there is an issue.

**Comment by melezhik** [comment]
Ok. Do you use any Sparrowdo:: modules for the moment? As I am not aware of
active users of Sparrowdo I am hesitant about fixing it asap. Sooner or
later I'll do it though.

**Comment by Altai-man** [comment]
I am not, but:
* I am preparing next Rakudo release and one of the thing I am doing is making sure that we have no regressions in (at least) installing a module. When a module a failing its tests, it just adds up some time for me to check what's the matter, if the compiler wrong, what can be done, etc.
* New potential users can check it up, see some breakage and become hesitant to look further (~weaklings die, no big deal!~).

In any case, it would be great to see this cool project being healthy. :)

**Comment by melezhik** [comment]
I've just released `Sparrowdo-Chef-Client-0.1.0.tar.gz` please see if the issue is gone.

**Comment by Altai-man** [comment]
It works now! :+1:
